fhevm/precompiles.go

```diff
@@ -2205,49 +2167,8 @@ func decryptRun(environment EVMEnvironment, caller common.Address, addr common.A
 }
 
 func decryptValue(environment EVMEnvironment, ct *tfheCiphertext) (uint64, error) {
-
-	logger := environment.GetLogger()
-	var fheType kms.FheType
-	switch ct.fheUintType {
-	case FheUint8:
-		fheType = kms.FheType_Euint8
-	case FheUint16:
-		fheType = kms.FheType_Euint16
-	case FheUint32:
-		fheType = kms.FheType_Euint32
-	}
-
-	// TODO: generate merkle proof for some data
-	proof := &kms.Proof{
-		Height:              4,
-		MerklePatriciaProof: []byte{},
-	}
-
-	decryptionRequest := &kms.DecryptionRequest{
-		FheType:    fheType,
-		Ciphertext: ct.serialization,
-		Request:    []byte{}, // TODO: change according to the structure of `Request`
-		Proof:      proof,
-	}
-
-	conn, err := grpc.Dial(kms.KmsEndpointAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
-	if err != nil {
-		return 0, errors.New("kms unreachable")
-	}
-	defer conn.Close()
-
-	ep := kms.NewKmsEndpointClient(conn)
-
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
-	defer cancel()
-
-	res, err := ep.Decrypt(ctx, decryptionRequest)
-	if err != nil {
-		logger.Error("decrypt failed", "err", err)
-		return 0, err
-	}
-
-	return uint64(res.Plaintext), err
+	v, err := ct.decrypt()
+	return v.Uint64(), err
```

Comment:
copy-paste from https://github.com/zama-ai/go-ethereum/blob/f24d3e742ec643b848e69b6daab671dc787f7d5d/core/vm/contracts.go#L2188-L2203.

FYI, optimistic requires are already checked for decrypt on L2146